src/utils/preprocessRequirement.ts

```diff
@@ -38,18 +38,24 @@ const preprocessRequirement = (requirement: Partial<Requirement>): Requirement =
   }
 
   if (processedRequirement.type?.startsWith("COVALENT_")) {
-    if (!processedRequirement?.data?.timestamps?.minAmount) {
+    if (
+      processedRequirement.data &&
+      !processedRequirement?.data?.timestamps?.minAmount
```

Comment:
Since we already check if `processedRequirement.data` is defined, I think we don't need optional chaining here, right? So I'd write something like this: `processedRequirement.data.timestamps?.minAmount`. Same goes for the if statement in line 48.